tests/data/pullback_metric_data.py

```diff
@@ -1,220 +1,574 @@
+import abc
+
 import geomstats.backend as gs
+from geomstats.geometry.euclidean import Euclidean
+from geomstats.geometry.manifold import Manifold
 from geomstats.geometry.pullback_metric import PullbackMetric
 from tests.data_generation import TestData
 
 
+class ImmersedSet(Manifold, abc.ABC):
```

Comment:
Why is this here?

Reply:
The main reason is that I feel it is still very experimental to move it to the main code. Nonetheless, it may be actually relevant to have access to it already (e.g. it will simplify one of the notebooks; someone may actually want to use it), so I'll move it it. We need to design it a bit better later though.